components/menu/menu.tsx

```diff
@@ -0,0 +1,203 @@
+        );
+    }
+
+    function handleAnchorButtonClick(event: MouseEvent<HTMLButtonElement>) {
+        event.preventDefault();
+
+        setAnchorElement(event.currentTarget);
+    }
+
+    function rendertriggerElement() {
+        // Since the open and close state lies in this component, we need to force the visibility of the trigger element
+        const forceVisibleOnOpen = isMenuOpen ? {display: 'block'} : undefined;
+
+        const triggerElement = (
+            <button
+                id={props.triggerId}
+                aria-controls={isMenuOpen ? props.menuId : undefined}
+                aria-haspopup='true'
+                aria-expanded={isMenuOpen ? 'true' : undefined}
+                aria-label={props.triggerAriaLabel}
+                className={props.triggerClassName}
+                onClick={handleAnchorButtonClick}
+                style={forceVisibleOnOpen}
+                tabIndex={0}
+            >
+                {props.triggerElement}
+            </button>
+        );
+
+        if (props.triggerTooltipText) {
+            return (
+                <OverlayTrigger
+                    delayShow={OVERLAY_TIME_DELAY}
+                    placement={props?.triggerTooltipPlacement ?? 'top'}
+                    overlay={
+                        <Tooltip
+                            id={props.triggerTooltipId}
+                            className={props.triggerTooltipClassName}
+                        >
+                            {props.triggerTooltipText}
+                        </Tooltip>
+                    }
+                    disabled={!props.triggerTooltipText || isMenuOpen}
+                >
+                    {triggerElement}
+                </OverlayTrigger>
+            );
+        }
+
+        return triggerElement;
+    }
+
+    function handleMenuClose(event: MouseEvent<HTMLDivElement | HTMLUListElement>) {
+        event.preventDefault();
+        setAnchorElement(null);
+    }
+
+    return (
+        <CompassDesignProvider theme={theme}>
+            {rendertriggerElement()}
+            <MuiMenuStyled
+                id={props.menuId}
+                anchorEl={anchorElement}
+                open={isMenuOpen}
+                onClose={handleMenuClose}
+                aria-label={props.menuAriaLabel}
+                elevation={24}
+            >
+                <MuiMenuList
+                    aria-labelledby={props.triggerId}
+                    onClick={handleMenuClose}
+                >
+                    {props.children}
+                </MuiMenuList>
+            </MuiMenuStyled>
+        </CompassDesignProvider>
+    );
+}
+
+const MuiMenuStyled = muiStyled(MuiMenu)<MuiMenuProps>(() => ({
+    '& .MuiPaper-root': {
+        backgroundColor: 'var(--center-channel-bg)',
+        boxShadow: 'var(--elevation-4) , 0 0 0 1px rgba(var(--center-channel-color-rgb), 0.08) inset',
+        minWidth: '114px',
+        maxWidth: '496px',
+        maxHeight: '80vh',
```

Comment:
Are these min/max dimensions still needed? I would've figured MUI could do this for us

Reply:
yes, these are Design/UX restrictions as per spec